### PR TITLE
Add Era modifier for Run 3 HLT conditions

### PR DESCRIPTION
**Pull request by davidwalter2**
#### PR description:

Currently, the era modifier "run2_HLTconditions_2018" is used when running on Run 3 data which defines the wrong trigger filter key for the ZCounting module. This PR fixes this issue such that the correct trigger filter key is taken.

- Added a new era modifier for the Run 3 HLT conditions in the new file [Configuration/Eras/python/Modifier_run3_HLTconditions_cff.py](https://github.com/cms-sw/cmssw/compare/master...davidwalter2:cmssw:master#diff-2ee612d4d87e28128c82e78bd57504f1e8fdf88e98a8cbcf520cc9327cbbab60)

- Use this era modifier for Run 3 in [Configuration/Eras/python/Era_Run3_cff.py](https://github.com/cms-sw/cmssw/compare/master...davidwalter2:cmssw:master#diff-bdce97c132f2495fedca6f8c58fe085659f25973341a407fff38d404b804c9bc)

- Use this era modifier in the Z counting configuration [DQMOffline/Lumi/python/ZCounting_cff.py](https://github.com/cms-sw/cmssw/compare/master...davidwalter2:cmssw:master#diff-e95da84091cdbbe20566f45e73617b01dd6105fe396b715be347ac5c9e7e1149)

#### PR validation:

The updates were tested in CMSSW_12_4 using the cmsDriver.py 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This PR is meant to be backported to CMSSW_12_4_X and CMSSW_12_5_X

### CODE REVIEW

File: Configuration/Eras/python/Era_Run3_cff.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 from Configuration.Eras.Modifier_run3_nanoAOD_devel_cff import run3_nanoAOD_devel
 from Configuration.Eras.Modifier_run3_egamma_cff import run3_egamma
 from Configuration.Eras.Modifier_run3_RPC_cff import run3_RPC
+from Configuration.Eras.Modifier_run3_HLTconditions_cff import run3_HLTconditions
 
-Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017, ctpps_2018]), run3_common, run3_egamma, run3_GEM, run3_HB, run3_HFSL, stage2L1Trigger_2021, ctpps_2022, dd4hep, run3_nanoAOD_devel, run3_RPC)
+Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017, ctpps_2018]), run3_common, run3_egamma, run3_GEM, run3_HB, run3_HFSL, stage2L1Trigger_2021, ctpps_2022, dd4hep, run3_nanoAOD_devel, run3_RPC, run3_HLTconditions)
```
Comment (makortel) [inline Configuration/Eras/python/Era_Run3_cff.py:18]: I believe the `run2_HLTconditions_2018` would need to be excluded here
```suggestion
Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017, ctpps_2018, run2_HLTconditions_2018]), run3_common, run3_egamma, run3_GEM, run3_HB, run3_HFSL, stage2L1Trigger_2021, ctpps_2022, dd4hep, run3_nanoAOD_devel, run3_RPC, run3_HLTconditions)
```

Comment (davidwalter2) [reply Configuration/Eras/python/Era_Run3_cff.py:18]: Sure, this can be done. What about other modules using the ```run2_HLTconditions_2018``` ? 

I see it is used in [DQMOffline/Trigger/python/TopMonitoring_cff.py](https://github.com/cms-sw/cmssw/blob/master/DQMOffline/Trigger/python/TopMonitoring_cff.py) and [DQMOffline/Trigger/python/BTagAndProbeMonitoring_cff.py](https://github.com/cms-sw/cmssw/blob/master/DQMOffline/Trigger/python/BTagAndProbeMonitoring_cff.py) but excluding it should have no impact because the first one has the same default configuration and the second one only imports it but does not make use of it. 

Now I see that  ```run2_HLTconditions_2018``` is also used in [Configuration/StandardSequences/python/Eras.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/StandardSequences/python/Eras.py#L84). I would add the ```run3_HLTconditions``` there, too. 

And should we do something here [PhysicsTools/NanoAOD/python/triggerObjects_cff.py](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L305) ?

Comment (missirol) [reply Configuration/Eras/python/Era_Run3_cff.py:18]: I'm no expert of Eras (nor polices related to them), but I know the HLT menu for pp collisions in recent CMSSW releases is one and is for Run 3 (ignoring developments related to Phase 2, which are irrelevant for this PR).

My vague recollection of these Run-2 modifiers (and maybe I'm wrong) is that they were related to Run-2 UL MCs, where the (offline-)DQM step would be done in a single release (10_6_X) for 2016+2017+2018, while the HLT step was done in different releases for the 3 different years; this meant that the HLT-DQM in 10_6_X needed to be able to adapt to different HLT Paths (from different menus).

Going back to the present (12_4_X and 12_5_X), the UL use case does not apply, we have the Run-3 HLT in the release, and my understanding is that its clients (DQM, nanoAOD, etc) should simply be configured accordingly.

In short, I don't see the need to have `run3_HLTconditions` (instead, I'd configure `zcounting` directly for the Run-3 HLT in the `cff`, which should be named `cfi` by the way), while indeed the ` run2_HLTconditions_2018` should be removed from the `Run3` era.

But if I'm missing something, please let me know.

Comment (davidwalter2) [reply Configuration/Eras/python/Era_Run3_cff.py:18]: Many thanks for the reminder. I agree that the better option is simply removing the ```run2_HLTconditions_2018``` from the ```Run3```. I will do this in another PR.

**Comment by missirol** [comment]
@davidwalter2 

Interestingly, this PR now is labelled as merged somehow. Next time, it might be best to just close it.

**Comment by davidwalter2** [comment]
> @davidwalter2
> 
> Interestingly, this PR now is labelled as merged somehow. Next time, it might be best to just close it.

Yes, I noticed the same. I reset the changes on my forked repository and through this, I think the PR was updated. Without any change, it was automatically merged. I really should improve my approach and branch out before doing the PR ... then I can also close it in the future, thanks!

**Comment by emanueleusai** [comment]
+1

- approving to remove it from dqm PR tracking system

**Comment by emanueleusai** [comment]
+dqm
